src/lib/lspace/layout/lalloc.rs

```diff
@@ -0,0 +1,768 @@
+    /// `space_between` : the amount of space to insert between child elements
+    /// `ref_point_index` : an optional index that identifies a child element that is to have its
+    /// reference point aligned with that of the region
+    pub fn alloc_linear(child_reqs: &Vec<&LReq>, child_allocs: &mut Vec<&mut LAlloc>,
+                        region_req: &LReq,
+                        region_pos: f64, region_size: f64, region_ref: Option<f64>,
+                        space_between: f64, ref_point_index: Option<usize>) {
+        let n = child_reqs.len();
+        assert_eq!(n, child_allocs.len());
+
+        // There are a few valid, consistent configurations that are identified and handled
+        match (ref_point_index, region_req.size(), region_ref) {
+            (None, &LNatSize::Size{..}, None) => {
+                // NO REFERENCE POINTS
+                match region_req.flex() {
+                    &LFlex::Fixed => {
+                        LAlloc::_alloc_children_linear(n, child_reqs, child_allocs, region_pos,
+                            space_between);
+                    },
+                    &LFlex::Flex{..} => {
+                        let natural_size = region_req.size().size();
+                        let minimum_size = natural_size - region_req.flex().shrink();
+
+                        LAlloc::_alloc_children(n, &child_reqs[..],
+                                                &mut child_allocs[..],
+                                                natural_size, minimum_size,
+                                                region_pos, region_size,
+                                                region_req.flex().stretch(),
+                                                space_between, false);
+                    },
+                }
+            },
+
+            (Some(ref_point_n), &LNatSize::Ref{..}, Some(region_before_ref)) => {
+                assert!(ref_point_n < n);
+                // REFERENCE POINT INDEX SPECIFIED
+                // REQUISITION HAS REF POINT
+                // REGION HAS REF POINT
+                match region_req.flex() {
+                    &LFlex::Fixed => {
+                        // Compute the offset required to position the array of children so that
+                        // the required reference point aligns with the allocated reference point.
+                        // Assuming that the region requisition has its reference point
+                        // aligned with that of the correct child, this should be fine...
+                        let before_offset = fast_max(
+                            region_before_ref - region_req.size().before_ref(), 0.0);
+                        LAlloc::_alloc_children_linear(n, &child_reqs[..],
+                            &mut child_allocs[..],
+                            before_offset + region_pos, space_between);
+                    },
+                    &LFlex::Flex{..} => {
```

Comment:
Fraid not. When an enum value has structure members you have to mention them otherwise rust complains.